### PR TITLE
feat: add service for storing merkle tree CAR files to S3

### PR DESCRIPTION
**Pull request by stbrody**
Not yet tested

### CODE REVIEW

File: src/services/merkle-car-service.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    return this.cars.get(anchorProofCID.toString()) || null
+  }
+}
+
+const S3_STORE_PATH = '/ceramic/cas/anchor/merkle-car/'
+const carFactory = new CARFactory()
+carFactory.codecs.add(DAG_JOSE)
+
+export class S3MerkleCarService {
+  readonly s3store: LevelUp.LevelUp = new LevelUp(new S3LevelDOWN(S3_STORE_PATH, new AWSSDK.S3()))
```
Comment (ukstv) [inline src/services/merkle-car-service.ts:35]: How is it configured? I mean, I would expect some credentials and a name of s3 bucket. Does that happens automagically somehow?

Comment (stbrody) [reply src/services/merkle-car-service.ts:35]: Yeah, my understanding is that the S3 library knows to look for certain specific environment variables and gets its configuration from that.  I believe that's how we usually configure the S3 settings for our Ceramic nodes currently.  @smrz2001 @3benbox can you confirm?

**Comment by stbrody** [comment]
Added @smrz2001 to the review to check out the S3 configuration

File: config/env/test.json [hunk 2]
```diff
@@ -5,6 +5,10 @@
   "readyRetryIntervalMS": 10000,
   "requireAuth": false,
   "schedulerIntervalMS": 10000,
+  "carStorage": {
+    "mode": "s3",
+    "s3BucketName": "myS3Bucket"
```
Comment (stbrody) [inline config/env/test.json:10]: is there a better default to use here?  Maybe we should put whatever we are actually using in our infra?  @smrz2001

File: src/services/merkle-car-service.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  async retrieveCarFile(anchorProofCID: CID): Promise<CAR | null> {
+    return this.cars.get(anchorProofCID.toString()) || null
+  }
+}
+
+const S3_STORE_SUFFIX = '/cas/anchor/merkle-car/'
+const carFactory = new CARFactory()
+carFactory.codecs.add(DAG_JOSE)
+
+export class S3MerkleCarService {
```
Comment (ukstv) [inline src/services/merkle-car-service.ts:35]: @stbrody Supposedly this implements `IMerkleCarService`

File: src/app.ts [hunk 4]
```diff
@@ -38,6 +38,7 @@ import { AnchorRequestParamsParser } from './ancillary/anchor-request-params-par
 import { HealthcheckService, IHealthcheckService } from './services/healthcheck-service.js'
 import { RequestService } from './services/request-service.js'
 import { AnchorBatchSqsQueueService } from './services/queue/sqs-queue-service.js'
+import { InMemoryMerkleCarService, S3MerkleCarService } from './services/merkle-car-service.js'
 
 type DependenciesContext = {
   config: Config
```
Comment (ukstv) [inline src/app.ts:44]: Please, add `IMerkleCarService` to `DependenciesContext` @stbrody

File: config/env/test.json (hunk 2 above)
Comment (smrz2001) [reply config/env/test.json:10]: Yeah, we use `ceramic-${env}-cas` so we could use `ceramic-dev-cas` here.

File: config/default.json [hunk 5]
```diff
@@ -14,6 +14,10 @@
   "readyRetryIntervalMS": 300000,
   "requireAuth": false,
   "schedulerIntervalMS": 300000,
+  "carStorage": {
+    "mode": "inmemory",
+    "s3BucketName": "myS3Bucket"
```
Comment (smrz2001) [inline config/default.json:19]: Can also use the default name suggested below.

Is it possible to parameterize this and use something like `ceramic-${env}-cas` here?

File: src/app.ts (hunk 4 above)
Comment (ukstv) [reply src/app.ts:44]: okay, disregard this one. Does not matter really.

File: src/services/merkle-car-service.ts (hunk 3 above)
Comment (ukstv) [reply src/services/merkle-car-service.ts:35]: Again, please disregard the comment.

**Pull request merged**